plotnine/geoms/annotation_logticks.py

```diff
@@ -0,0 +1,139 @@
+import math
+import warnings
+
+import numpy as np
+import pandas as pd
+import mizani.transforms
+
+from ..doctools import document
+from ..coords import coord_flip
+from ..exceptions import PlotnineWarning
+from .annotate import annotate
+from .geom_rug import geom_rug
+
+
+class _geom_logticks(geom_rug):
+    """internal geom implementing drawing of annotation_logticks"""
+
+    DEFAULT_AES = {}
+    DEFAULT_PARAMS = {
+        'stat': 'identity',
+        'position': 'identity',
+        'na_rm': False,
+        'sides': 'bl',
+        'alpha': 1,
+        'color': 'black',
+        'size': 0.5,
+        'linetype': 'solid',
+        'length': (0.036, 0.0225, 0.012),
+    }
+    legend_geom = 'path'
+
+    @staticmethod
+    def _check_log_scale(sides, scales, is_coord_flip):
+        x_is_log_10 = isinstance(scales.x.trans, mizani.transforms.log10_trans)
+        y_is_log_10 = isinstance(scales.y.trans, mizani.transforms.log10_trans)
+        if is_coord_flip:
+            x_is_log_10, y_is_log_10 = y_is_log_10, x_is_log_10
+        result = []
+        if (not x_is_log_10 and ('t' in sides or 'b' in sides)):
+            warnings.warn("annotation_logticks for x-axis,"
+                          " but x-axis is not log10 - nonsensical marks show.",
+                          PlotnineWarning)
+        if (not y_is_log_10 and ('l' in sides or 'r' in sides)):
+            warnings.warn("annotation_logticks for y-axis,"
+                          " but y-axis is not log10 - nonsensical marks show.",
+                          PlotnineWarning)
+        return result
+
+    @classmethod
+    def draw_panel(cls, ignored_data, panel_params, coord, ax, **params):
+        if 'l' in params['sides'] or 'r' in params['sides']:
+            ymin, ymax = panel_params['y_range']
+            cls.draw_group('y', ymin, ymax, panel_params, coord, ax, params)
+        if 'b' in params['sides'] or 't' in params['sides']:
+            xmin, xmax = panel_params['x_range']
+            cls.draw_group('x', xmin, xmax, panel_params, coord, ax, params)
+
+    @staticmethod
+    def _calc_ticks(lower_log, upper_log, marks_at):
```

Comment:
It is better of `_calc_ticks` just take a range. Maybe something like:

```python
@staticmethod
def _calc_ticks(range):
    
    def _calc(positions):
        ...

    major = _calc([1.0])
    middle = _calc([0.5])
    minor = _calc([.1, 0.2, 0.3, 0.4, 0.6, 0.7, 0.8, 0.9])
    return major, middle, minor
```

Where the `(major, middle, minor)` tick positions are already log transformed. This has all the calculations happen in one appropriately named function.

Reply:
I've refactored accordingly, but the inner _calc is till a class level (static) function so it can be tested.